pkg/utils/resources/resources.go

```diff
@@ -24,27 +24,37 @@ import (
 // RequestsForPods returns the total resources of a variadic list of podspecs.
 func RequestsForPods(pods ...*v1.Pod) v1.ResourceList {
 	var resources []v1.ResourceList
+	var init []v1.ResourceList
 	for _, pod := range pods {
 		for _, container := range pod.Spec.Containers {
-			resources = append(resources, container.Resources.Requests)
+			resources = append(resources, MergeResourceLimitsIntoRequests(container))
+		}
+		for _, container := range pod.Spec.InitContainers {
+			init = append(init, MergeResourceLimitsIntoRequests(container))
 		}
 	}
-	merged := Merge(resources...)
-	merged[v1.ResourcePods] = *resource.NewQuantity(int64(len(pods)), resource.DecimalExponent)
-	return merged
+	containerResources := Merge(resources...)
+	initResources := MaxResources(init...)
```

Comment:
Oops, I don't think this logic is right.

Reply:
For each pod we need to:
1) Sum up non-InitContainer resources for the pod
2) Find the max resources for each resource type for InitContainers for the pod
3) Compare sum of "Container" resources against max of "InitContainer" to get the max resources required for the pod.

(RequestForPods then sums up the max resources for all of the pods.)

New revision incoming.

Reply:
This is corrected in the most recent revision.